lib/instream/connection/query_runner.ex

```diff
@@ -28,40 +53,90 @@ defmodule Instream.Connection.QueryRunner do
   def read(%Query{} = query, opts, conn) do
     headers = conn |> Headers.assemble()
     url     =
-         conn
+      conn
       |> URL.query()
       |> URL.append_database(opts[:database])
       |> URL.append_epoch(query.opts[:precision])
       |> URL.append_query(query.payload)
 
     http_opts = Keyword.get(conn, :http_opts, [])
 
-    { :ok, status, headers, client } = :hackney.get(url, headers, "", http_opts)
-    { :ok, response }                = :hackney.body(client)
+    { query_time, { :ok, status, headers, client }} = :timer.tc fn ->
+      :hackney.get(url, headers, "", http_opts)
+    end
+
+    { :ok, response } = :hackney.body(client)
+    result            =
+      { status, headers, response }
+      |> Response.maybe_parse(opts)
+
+    if false != opts[:log] do
```

Comment:
I wanted to force the decision of deactivating the logging to the user by requiring a fixed value. Basically just an arbitrary design choice, but this way I don't need to keep track of default values mixing in with the behaviour.